Chapter One FR - first pass (#1)
Scttpr/You-Dont-Know-JS: get-started/ch1.md

Une fois qu'une proposition atteint l'étape 4, elle est éligible pour être incluse à la prochaine révision annuelle du langage. Cela peut prendre de quelques mois à quelques années pour qu'une proposition passe toutes les étapes.

Toutes les propositions sont gérées publiquement, sur le dépôt Github du TC39 : https://github.com/tc39/proposals

Toute personne, du TC39 ou non, est encouragée à participer à ces discussions publiques et aux différentes étapes de travail sur ces propositions. Cependant, seul les membres du TC39 peuvent assister aux réunions et voter les propositions et changements. Donc, de ce fait, la voix d'un membre du TC39 aura un impact important sur la direction que suivra JS.

Contrairement à des mythes établis et malheureusement perpétués, il n'y a *pas* de multiples versions de JavaScript dans la nature. Il n'y a qu'**un JS**, le standard officiel maintenu par TC39 et ECMA.

> Au début des années 2000, quand Microsoft a maintenu une fourche et une version rétro-ingénieurisé (et pas totalement compatible) de JS appelée "JScript", il y a avait légitimement "plusieurs version" de JS. Mais ces jours sont loin. Cela n'est donc plus d'actualité et imprécis d'affirmer cela aujourd'hui.

Reviewer: ```suggestion
Au début des années 2000, quand Microsoft a maintenu une fourche et une version rétro-ingénieurisée (et pas totalement compatible) de JS appelée "JScript", il y avait légitimement "plusieurs versions" de JS. Mais ces jours sont révolus, on ne peut donc plus parler de ça désormais.
```

"Fourche", je vois pas comment le traduire...

# You Don't Know JS Yet: Get Started - 2nd Edition
# Chapitre 1 : Qu'est-ce que JavaScript ?

Vous ne connaissez pas encore JavaScript. Tout comme moi, du moins pas complètement. D'ailleurs comme aucun de nous. Cependant nous pouvons tous faire plus ample connaissance avec JavaScript.

> Dans ce premier chapitre du premier livre de la série *You Don't Know JS Yet* (YDKJSY), nous allons prendre le temps de construire de solides bases pour préparer la suite. Nous allons donc commencer par explorer toute une variété d'importants détails de fond, d'effacer quelques mythes et idées fausses à propos de ce qu'est réellement le langage (et ce qu'il n'est pas).

Reviewer: idées fausses: "fausses idées" ou "idées reçues" sera plus joli

C'est un aperçu fondamental dans l'identité et la façon dont Javascript est organisé et maintenu que tous les développeurs JS devrait comprendre. Si vous voulez faire plus amples connaissance avec JS, voici la première étape de votre voyage : débuter.

<!-- In this first chapter of the first book of the *You Don't Know JS Yet* (YDKJSY) series, we will take some time to build a foundation to move forward on. We need to start by covering a variety of important background housekeeping details, clearing up some myths and misconceptions about what the language really is (and isn't!). -->

<!-- This is valuable insight into the identity and process of how JS is organized and maintained; all JS developers should understand it. If you want to get to know JS, this is how to *get started* taking the first steps in that journey. -->

## A propos du livre

> Je souligne le mot voyage car *connaître JS* n'est pas une destination, c'est une direction. Quel que soit le temps passé avec le langage, vous allez toujours trouver quelque chose à apprenddre pour le comprendre un peu mieux. Donc ne percevez pas ce livre comme quelque chose à rusher pour une prouesse de rapidité. C'est plutôt dans la patience et la persistence que vous ferez les meilleurs premiers petits pas.

Reviewer: ```suggestion
Je souligne le mot voyage car *connaître JS* n'est pas une destination, c'est une direction. Quel que soit le temps passé avec le langage, vous allez toujours trouver quelque chose à aprenddre pour le comprendre un peu mieux. Donc ne percevez pas ce livre comme quelque chose à rusher pour une prouesse de rapidité. C'est plutôt dans la patience et la persistence que vous ferez les meilleurs premiers petits pas.
```

<!-- I emphasize the word journey because *knowing JS* is not a destination, it's a direction. No matter how much time you spend with the language, you will always be able to find something else to learn and understand a little better. So don't look at this book as something to rush through for a quick achievement. Instead, patience and persistence are best as you take these first few steps. -->

A la suite de ce chapitre de fond, le reste du livre présente un panomara de haut-niveau de ce que vous trouverez en creusant et en étudiant le JS dans ces livres.

<!-- Following this background chapter, the rest of the book lays out a high-level map of what you will find as you dig into and study JS with the YDKJSY books. -->

> En particulier, le chapitre 4 identifie trois piliers principaux autour duquel le JS est organisé: portée/closure, prototypes:objets et types/coercition. JS est un langage vaste et sophistiqué, avec beaucoup de fonctionnalités et capacités. Mais tout le JS est fondé sur ces trois piliers fondateurs.

Reviewer: ```suggestion
Le chapitre 4, notamment, identifie trois piliers principaux autour duquel le JS est organisé: portée/closure, prototypes/objets et types/coercition. JS est un langage vaste et sophistiqué, avec beaucoup de fonctionnalités et capacités. Mais tout le JS est fondé sur ces trois piliers fondateurs.
```

<!-- In particular, Chapter 4 identifies three main pillars around which the JS language is organized: scope/closures, prototypes/objects, and types/coercion. JS is a broad and sophisticated language, with many features and capabilities. But all of JS is founded on these three foundational pillars. -->

> Gardez à l'esprit que même si ce livre est intitulé "Débuter", il n'est pas destiné à être un livre d'introduction. Le travail principal de ce livre est de vous préparer à étudier JS profondément tout au long de la série. Il est écrit en assumant que vous êtes déjà familié avec JS avec déjà plusieurs mois d'experience. Donc pour en tirer le plus grand bénéfice, soyez certain d'avoir déjà passé beaucoup de temps à écrire du code JS pour construire votre expérience.

Reviewer: ```suggestion
Gardez à l'esprit que même si ce livre est intitulé "Débuter", il n'est pas destiné à être un livre d'introduction. Le travail principal de ce livre est de vous préparer à étudier JS profondément tout au long de la série. Il est écrit en supposant que vous êtes déjà familié avec JS avec déjà plusieurs mois d'experience. Donc pour en tirer le plus grand bénéfice, soyez certain d'avoir déjà passé beaucoup de temps à écrire du code JS pour construire votre expérience.
```

Même si vous avez déjà écrit beaucoup de JS, ce livre ne devrait pas être efleuré ni sauté. Prenez votre temps pour parcourir l'ensemble de son contenu : **un bon début dépend toujours d'un solide premier pas.**

<!-- Even if you've already written a lot of JS before, this book should not be skimmed over or skipped; take your time to fully process the material here. **A good start always depends on a solid first step.** -->

## Qu'y a-t-il avec ce nom ?

Le nom JavaScript est probablement le plus mal compris des noms de langages de programmation.

<!-- The name JavaScript is probably the most mistaken and misunderstood programming language name. -->

Est-il lié à Java ? Est-ce uniquement la version scriptée de Java ? Est-ce uniquement pour écrire des scripts et non de réels programmes ?

<!-- Is this language related to Java? Is it only the script form for Java? Is it only for writing scripts and not real programs? -->

> La vérité est que le nom JavaScript est un artefact une combine marketing. Quand Brendan Eich a d'abord conçu le langage, il l'a appelé Mocha. En interne à Netscape, la marque LiveScript était utilisée. Mais quand il a fallu le nommé publiquement, "JavaScript" remporta les suffrages.

Reviewer: ```suggestion
En fait, le nom JavaScript est un artefact de magouilles marketing. Quand Brendan Eich a conçu le langage, il l'a appelé Mocha. En interne, à Netscape, la marque LiveScript était utilisée. Mais quand il a fallu le nommer publiquement, "JavaScript" remporta les suffrages.
```

<!-- The truth is, the name JavaScript is an artifact of marketing shenanigans. When Brendan Eich first conceived of the language, he code-named it Mocha. Internally at Netscape, the brand LiveScript was used. But when it came time to publicly name the language, "JavaScript" won the vote. -->

> Pourquoi ? Car le langage est originellement designé pour attirer une audience principalement composée de développeurs Java et parce que le mot "script" était populaire à ce moment, faisant référent à des programmes très légers. Ces programmes ultra légers seraient les premier à s'insérer dans les pages de cette nouvelle chose appelée web !

Reviewer: ```suggestion
Pourquoi ? Car le langage est originellement désigné pour attirer une audience principalement composée de développeurs Java et parce que le mot "script" était à ce moment là populaire, faisant référence à des programmes très légers. Ces programmes ultra légers seraient les premier à s'insérer dans les pages de cette nouvelle chose appelée web !
```

Author: Here I changed a bit more but thanks for suggestions !

<!-- Why? Because this language was originally designed to appeal to an audience of mostly Java programmers, and because the word "script" was popular at the time to refer to lightweight programs. These lightweight "scripts" would be the first ones to embed inside of pages on this new thing called the web! -->

> En d'autre termes, JavaScript était un stratagème marketing pour positionner le langage comme une bonne alternative pour écrire le plus lourd et bien connu Java de cette époque. C'est pour cette raison qu'il aurait tout aussi bien pu s'appeler "WebJava".

Reviewer: ```suggestion
En d'autre termes, JavaScript était un stratagème marketing pour essayer de positionner ce langage comme une alternative acceptable à l'écriture du Java, plus lourd et plus connu à l'époque. C'est pour cette raison qu'il aurait tout aussi bien pu s'appeler "WebJava".
```

<!-- In other words, JavaScript was a marketing ploy to try to position this language as a palatable alternative to writing the heavier and more well-known Java of the day. It could just as easily have been called "WebJava," for that matter. -->

Il y a quelques similitudes superficielles entre le code JavaScript et le code Java. Ces ressemblances ne proviennent pas forcément d'un développement partagé mais plutôt d'une volonté des deux langages de viser les développeurs avec des attentes assummées de syntaxe par rapport au C (et par extension au C++).

<!-- There are some superficial resemblances between JavaScript's code and Java code. Those similarities don't particularly come from shared development, but from both languages targeting developers with assumed syntax expectations from C (and to an extent, C++). -->

> Par exemple on utilise le `{` pour commencer un bloc de code et `}` pour le terminer, comme en C/C++ et Java. On utilise aussi le `;` pour ponctuer la fin d'une déclaration.

Reviewer: ```suggestion
Par exemple, on utilise le `{` pour commencer un bloc de code et `}` pour le terminer, comme en C/C++ et Java. On utilise aussi le `;` pour ponctuer la fin d'une déclaration.
```

<!-- For example, we use the `{` to begin a block of code and the `}` to end that block of code, just like C/C++ and Java. We also use the `;` to punctuate the end of a statement. -->

> D'une certaine manière, les relations légales sont plus profonde que la syntaxe. Oracle (via Sun), l'entreprise qui possède et maintient Java a aussi déposé la marque "JavaScript" (via Netscape). Cette marque n'est presque jamais appliquée et semble de pouvoir jamais l'être à l'heure d'aujourd'hui.

Reviewer: ```suggestion
D'une certaine manière, les relations légales sont plus profonde que la syntaxe. Oracle (via Sun), l'entreprise qui possède et maintient Java a aussi déposé la marque "JavaScript" (via Netscape). Cette marque n'est presque jamais appliquée et ne pourrait probablement jamais l'être.
```

<!-- In some ways, legal relationships run even deeper than the syntax. Oracle (via Sun), the company that still owns and runs Java, also owns the official trademark for the name "JavaScript" (via Netscape). This trademark is almost never enforced, and likely couldn't be at this point. -->

> Pour ces raisons, certains ont suggéré qu'on utilise JS au lien de JavaScript. C'est un raccourci très commun et même un bon candidat pour vendre officiellement le langage. En effet, ces livres utilisent presque exclusivement le terme JS pour faire référence au langage.

Reviewer: ```suggestion
Pour ces raisons, certains ont suggéré qu'on utilise JS au lieu de JavaScript. C'est un raccourci très commun et même un bon candidat pour vendre officiellement le langage. En effet, ces livres utilisent presque exclusivement le terme JS pour faire référence au langage.
```

<!-- For these reasons, some have suggested we use JS instead of JavaScript. That is a very common shorthand, if not a good candidate for an official language branding itself. Indeed, these books use JS almost exclusively to refer to the language. -->

Plus éloigné de la marque détenue par Oracle, le nom officiel du langage spécifié par le TC39 et formalisé par les standards ECMA est **ECMAScript**. En effet, depuis 2016, le nom officiel du langage a été suffixé par l'année de révision. Au moment d'écrire ces lignes, il s'agit de la version ECMAScript 2019, autrement abrégée ES2019.

<!-- Further distancing the language from the Oracle-owned trademark, the official name of the language specified by TC39 and formalized by the ECMA standards body is **ECMAScript**. And indeed, since 2016, the official language name has also been suffixed by the revision year; as of this writing, that's ECMAScript 2019, or otherwise abbreviated ES2019. -->

> En d'autres termes, le JavaScript/JS qui est éxéxuté par votre navigateur ou dans Node.js est *une* implémentation du standard ES2019

Reviewer: ```suggestion
En d'autres termes, le JavaScript/JS qui est éxécuté par votre navigateur ou dans Node.js est *une* implémentation du standard ES2019
```

<!-- In other words, the JavaScript/JS that runs in your browser or in Node.js, is *an* implementation of the ES2019 standard. -->

| NOTE: |
| :--- |

> | N'utilisez pas des termes comme "JS6" ou "ES8" pour faire référence au langage. Certain le font mais ces appelations contribuent à perpétuer la confusion. Vous devriez coller à "ES20xx" ou juste "JS" |

Reviewer: ```suggestion
| N'utilisez pas des termes comme "JS6" ou "ES8" pour faire référence au langage. Certain le font mais ces appelations contribuent à perpétuer la confusion. Tenez vous en à "ES20xx" ou juste "JS" |
```

<!-- | NOTE: |
| :--- |
| Don't use terms like "JS6" or "ES8" to refer to the language. Some do, but those terms only serve to perpetuate confusion. "ES20xx" or just "JS" are what you should stick to. | -->

Que vous l'appeliez JavaScript, JS, ECMAScript, ou ES2019, ce n'est définitivement pas une variante du langage Java !

<!-- Whether you call it JavaScript, JS, ECMAScript, or ES2019, it's most definitely not a variant of the Java language! -->

> "Java is to JavaScript as ham is to hamster." --Jeremy Keith, 2009

## Spécifications du langage

> J'ai mentionné le TC39, le comité de direction technique qui gère JS. Leur première tâche est de gérer les spécifications offcieielles du langage. Il se réunit régulièrement pour valider les changements agréés qu'ils soumettent ensuite à ECMA, l'organisation qui gère les standards.

Reviewer: ```suggestion
J'ai mentionné le TC39, le comité de direction technique qui gère JS. Leur première tâche est de gérer les spécifications officielles du langage. Il se réunit régulièrement pour valider les changements agréés qu'ils soumettent ensuite à ECMA, l'organisation qui gère les standards.
```

<!-- I mentioned TC39, the technical steering committee that manages JS. Their primary task is managing the official specification for the language. They meet regularly to vote on any agreed changes, which they then submit to ECMA, the standards organization. -->

La syntaxe JS et son comportement sont définis dans les spécifications ES.

<!-- JS's syntax and behavior are defined in the ES specification. -->

> ES2019 est la dixième spécification/révision majeure depuis l'origine de JS en 1995, donc dans l'adresse URL officielle des spécifications hébergée par ECMA, vous toruvertez "10.0".

Reviewer: ```suggestion
ES2019 est la dixième spécification/révision majeure depuis l'origine de JS en 1995, donc dans l'adresse URL officielle des spécifications hébergée par ECMA, vous trouverez "10.0".
```

https://www.ecma-international.org/ecma-262/10.0/

<!-- ES2019 happens to be the 10th major numbered specification/revision since JS's inception in 1995, so in the specification's official URL as hosted by ECMA, you'll find "10.0": -->

<!-- https://www.ecma-international.org/ecma-262/10.0/ -->

> Le comité TC39 est composé d'entre 50 et environ 100 différentes personnes provenant de différentes entreprises investies dans le web comme les développeurs de navigateurs (Mozilla, Google, Apple), les fabriquants d'appareils (Samsung, etc). Tous les membres sont volontaires bien que la plupart des employés de ces entreprises peuvent recevoir des compensations contre leur travail au sein du comité.

Reviewer: ```suggestion
Le comité TC39 est composé d'environ 50 à 100 personnes différentes provenant de diverses entreprises investies dans le web comme les développeurs de navigateurs (Mozilla, Google, Apple), les fabricants d'appareils (Samsung, etc). Tous les membres sont volontaires bien que la plupart des employés de ces entreprises peuvent recevoir des compensations contre leur travail au sein du comité.
```

<!-- The TC39 committee is comprised of between 50 and about 100 different people from a broad section of web-invested companies, such as browser makers (Mozilla, Google, Apple) and device makers (Samsung, etc). All members of the committee are volunteers, though many of them are employees of these companies and so may receive compensation in part for their duties on the committee. -->

> TC39 se réunit généralement tous les mois pour trois jours pour vérifier le travail réalisé par ses membres depuis le dernier meeting, pour discuter des problèmes et voter les propositions. Les lieux de réunions tourne entre les entreprises membre prête à accueillir l'événement.

Reviewer: ```suggestion
TC39 se réunit généralement tous les mois pendant trois jours pour vérifier le travail réalisé par ses membres depuis le précédent meeting, pour discuter des problèmes et voter les propositions. Les lieux de réunions tournent entre les entreprises membre prêtes à accueillir l'événement.
```

<!-- TC39 meets generally about every other month, usually for about three days, to review work done by members since the last meeting, discuss issues, and vote on proposals. Meeting locations rotate among member companies willing to host. -->

> Toutes les propositions du TC39 passent pas un processus découpé en cinq étapes. Comme nous sommes des développeurs, la première étape est le 0, de 0 à 4 donc. Vous pouvez trouver davatange d'informations ici : https://tc39.es/process-document/

Reviewer: ```suggestion
Toutes les propositions du TC39 passent pas un processus découpé en cinq étapes. Comme nous sommes des développeurs, la première étape est le 0, de 0 à 4 donc. Vous pouvez trouver davantage d'informations ici : https://tc39.es/process-document/
```

<!-- All TC39 proposals progress through a five-stage process—of course, since we're programmers, it's 0-based!—Stage 0 through Stage 4. You can read more about the Stage process here: https://tc39.es/process-document/ -->

> L'étape 0 signifie que quelqu'un au TC39 pense qu'une idée en vaut la peine et planifie de travailler sur le sujet. Cela signifie que beaucoup d'idées qui sont proposées par des non membres du TC39 via des réseaux informels comme les réseaux sociaux ou les billtes de blog sont vraiment des "pré-étape 0". Vous devez être un membre du TC39 pour soutenir une proposition pour être officiellement considéré comme étape 0.

Reviewer: ```suggestion
L'étape 0 signifie que quelqu'un au TC39 pense qu'une idée en vaut la peine et planifie de travailler sur le sujet. Cela signifie que beaucoup d'idées qui sont proposées par des non membres du TC39 via des réseaux informels, comme les réseaux sociaux ou les billets de blog, sont plutôt des "pré-étape 0". Vous devez être membre du TC39 pour que votre proposition soit officiellement considéré comme étape 0.
```

<!-- Stage 0 means roughly, someone on TC39 thinks it's a worthy idea and plans to champion and work on it. That means lots of ideas that non-TC39 members "propose," through informal means such as social media or blog posts, are really "pre-stage 0." You have to get a TC39 member to champion a proposal for it to be considered "Stage 0" officially. -->

> Une fois qu'une proposition atteint l'étape 4, elle est éligbile pour être incluse dans la prochaine révision anuelle du langage. Cela peut prendre de quelques mois à quelques années pour qu'une proposition passe toutes les étapes.

Reviewer: ```suggestion
Une fois qu'une proposition atteint l'étape 4, elle est éligible pour être incluse dans la prochaine révision annuelle du langage. Cela peut prendre de quelques mois à quelques années pour qu'une proposition passe toutes les étapes.
```

<!-- Once a proposal reaches "Stage 4" status, it is eligible to be included in the next yearly revision of the language. It can take anywhere from several months to a few years for a proposal to work its way through these stages. -->

Toutes les propositions sont gérées publiquement, sur le dépôt Github du TC39 : https://github.com/tc39/proposals

<!-- All proposals are managed in the open, on TC39's Github repository: https://github.com/tc39/proposals -->

> Toute personne, du TC39 ou non, est encouragée à participer à ses discussions publiques et au processus de travail sur ses propositions. Cependant, seul les membres du TC39 peuvent assister aux réunions et voter les propositions et changements. Donc, de ce fait, la voie d'un membre du TC39 porte beaucoup de poids sur les futures direction que JS prendra.

Reviewer: ```suggestion
Toute personne, du TC39 ou non, est encouragée à participer à ces discussions publiques et au processus de travail sur ces propositions. Cependant, seul les membres du TC39 peuvent assister aux réunions et voter les propositions et changements. De ce fait, la voix d'un membre du TC39 aura un impact important sur la direction que suivra JS.
```

<!-- Anyone, whether on TC39 or not, is welcome to participate in these public discussions and the processes for working on the proposals. However, only TC39 members can attend meetings and vote on the proposals and changes. So in effect, the voice of a TC39 member carries a lot of weight in where JS will go. -->

> Contrairement à des mythes établit et perpétué frustreusement, il n'y a *pas* de multiples versions de JavaScript dans la nature. Il n'y a qu'**un JS**, le standard officiel maintenu par le TC39 et ECMA.

Reviewer: ```suggestion
Contrairement à des mythes établis et désespérément perpétués, il n'y a *pas* de multiples versions de JavaScript dans la nature. Il n'y a qu'**un JS**, le standard officiel maintenu par le TC39 et ECMA.
```
